### PR TITLE
WIP: add fmu-v4 serial rc scan

### PR DESCRIPTION
**Pull request by kd0aij**
NOT flight tested yet
still needs ST24 and SUMD work, also DSM bind function
DSM and SBUS working on fmu-v4, needs testing on fmu-v2 to validate #ifdefs
also need to disable scanning once locked

### CODE REVIEW

**Comment by kd0aij** [comment]
scan disabled on lock, also verified that fmu-v2 still works with PPMsum input
probably should pull the HRT_PPM case into the RC scan block too

**Comment by LorenzMeier** [comment]
```
src/drivers/px4fmu/fmu.cpp bad formatting, please run "./Tools/fix_code_style.sh src/drivers/px4fmu/fmu.cpp"
```

**Comment by LorenzMeier** [comment]
@kd0aij Yes, the PPM case should be pulled in. Serial leads to an interrupt storm and we should de-configure the PPM pin to avoid that while its not in use.

File: src/drivers/px4fmu/fmu.cpp [hunk 1]
```diff
@@ -811,45 +809,143 @@ PX4FMU::cycle()
...
+		} else if (hrt_absolute_time() - rc_scan_last_lock < rc_scan_max
+				|| hrt_absolute_time() - rc_scan_begin < rc_scan_max) {
+
+			uint8_t n_bytes = 0;
+			uint8_t *bytes;
+			bool dsm_11_bit;
+			uint16_t raw_rc_values[input_rc_s::RC_INPUT_MAX_CHANNELS];
+			uint16_t raw_rc_count;
+
+			rc_updated = dsm_input(_rcs_fd, raw_rc_values, &raw_rc_count, &dsm_11_bit, &n_bytes, &bytes,
```
Comment (LorenzMeier) [inline src/drivers/px4fmu/fmu.cpp]: It becomes easier if you pull out the serial read and do the dsm_parse here. Then you can copy that pattern for SUMD and ST24 (read and parse) and call it done.

**Comment by LorenzMeier** [comment]
Awesome! I think its a huge win we have DSM input working before the board reaches users. It would be great to stabilize / test this with minimal effort and then put out a 1.1.2 release which can be used by Phill's customers.

**Comment by kd0aij** [comment]
Should we make serial RC input support mandatory in fmu.cpp?
It would allow getting rid of several #ifdef tests

**Comment by LorenzMeier** [comment]
No, because its only supported on Pixracer.

**Comment by LorenzMeier** [comment]
You can of course make it optional, but compiled in.

**Pull request closed without merging**